packages/cli/src/commands/help.ts

```diff
@@ -0,0 +1,212 @@
+const NEWLINE = '\n';
+const MAX_LINE_LENGTH = process.stdout.columns;
+
+export interface CommandOption {
+  name: string;
+  shorthand: string | null;
+  type: 'boolean' | 'string';
+  argument?: string;
+  deprecated: boolean;
+  description: string;
+  multi: boolean;
+}
+export interface CommandArgument {
+  name: string;
+  required: boolean;
+}
+export interface CommandExample {
+  name: string;
+  value: string | string[];
+}
+export interface Command {
+  name: string;
+  description: string;
+  arguments: CommandArgument[];
+  options: CommandOption[];
+  examples: CommandExample[];
+}
+
+function calcLineLength(line: string[]) {
+  return stripAnsi(lineToString(line)).length;
+}
+
+// Insert spaces in between non-whitespace items only
+function lineToString(line: string[]) {
+  let string = '';
+  for (let i = 0; i < line.length; i++) {
+    if (i === line.length - 1) {
+      string += line[i];
+    } else {
+      const curr = line[i];
+      const next = line[i + 1];
+      string += curr;
+      if (curr.trim() !== '' && next.trim() !== '') {
+        string += ' ';
+      }
+    }
+  }
+  return string;
+}
+
+function outputArrayToString(outputArray: string[]) {
+  return outputArray.reduce((acc, line) => acc + `${line}${NEWLINE}`, '');
+}
+
+/**
+ * Example: `▲ vercel deploy [path] [options]`
+ * @param command
+ * @returns
+ */
+function buildCommandSynopsisLine(command: Command) {
+  const line: string[] = [LOGO, chalk.bold(NAME), chalk.bold(command.name)];
+  if (command.arguments.length > 0) {
+    for (const argument of command.arguments) {
+      line.push(argument.required ? argument.name : `[${argument.name}]`);
+    }
+  }
+  if (command.options.length > 0) {
+    line.push('[options]');
+  }
+  return lineToString(line);
+}
+
+function buildCommandOptionLines(command: Command) {
+  // Filter out deprecated options.
+  command.options = command.options.filter(option => !option.deprecated);
+
+  // Initialize output array with header and empty line
+  const outputArray: string[] = [chalk.dim(`Options:`), ''];
+
+  // Start building option lines
+  const optionLines: string[][] = [];
+  // Sort command options alphabetically
+  command.options.sort((a, b) =>
+    a.name < b.name ? -1 : a.name > b.name ? 1 : 0
+  );
+  // Keep track of longest "start" of an option line to determine description spacing
+  let maxLineStartLength = 0;
+  // Iterate over options and create the "start" of each option (e.g. `  -b, --build-env <key=value>`)
+  for (const option of command.options) {
+    const startLine: string[] = [INDENT];
+    if (option.shorthand) {
+      startLine.push(`-${option.shorthand},`);
+    }
+    startLine.push(`--${option.name}`);
+    if (option.argument) {
+      startLine.push(`<${option.argument}>`);
+    }
+    // the length includes the INDENT
+    const lineLength = calcLineLength(startLine);
+    maxLineStartLength =
+      lineLength > maxLineStartLength ? lineLength : maxLineStartLength;
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏

```suggestion
    maxLineStartLength = Math.max(lineLength, maxLineStartLength);
```